picard/script/__init__.py

```diff
@@ -101,3 +111,229 @@ def enabled_tagger_scripts_texts():
+        Properties in the JSON string that are not found in the script object are ignored.
+
+        Args:
+            json_string (str): JSON string containing the property settings.
+
+        Returns:
+            object: An instance of the class, populated from the property settings in the JSON string.
+        """
+        new_object = cls()
+        new_object.update_from_json(json_string)
+        return new_object
+
+    def update_from_json(self, json_string):
+        """Updates the values of the properties based on the contents of the JSON string provided.
+        Properties in the JSON string that are not found in the script object are ignored.
+
+        Args:
+            json_string (str): JSON string containing the property settings.
+        """
+        settings = json.loads(json_string)
+        self._update_from_dict(settings)
+
+
+class FileNamingScript(PicardScript):
+    """Picard file naming script class
+    """
+    TYPE = PicardScriptType.FileNamingScript
+    JSON_OUTPUT = ['title', 'script', 'author', 'description', 'license', 'version', 'last_updated']
+
+    def __init__(self, script='', title='', id=None, readonly=False, deletable=True, author='', description='', license='', version='', last_updated=None):
+        """Creates a Picard file naming script object.
+
+        Args:
+            script (str): Text of the script.
+            title (str): Title of the script.
+            id (str): ID code for the script. Defaults to a system generated uuid.
+            readonly (bool): Identifies if the script is readonly. Defaults to False.
+            deletable (bool): Identifies if the script can be deleted from the selection list. Defaults to True.
+            author (str): The author of the script. Defaults to ''.
+            description (str): A description of the script, typically including type of output and any required plugins or settings. Defaults to ''.
+            license (str): The license under which the script is being distributed. Defaults to ''.
+            version (str): Identifies the version of the script. Defaults to ''.
+            last_updated (str): The UTC date and time when the script was last updated. Defaults to current date/time.
+        """
+        super().__init__(script=script, title=title, id=id, last_updated=last_updated)
+        self.readonly = readonly    # for presets
+        self.deletable = deletable  # Allow removal from list of scripts
+        self.author = author
+        self.description = description
+        self.license = license
+        self.version = version
+
+    def copy(self):
+        new_object = super().copy()
+        new_object.readonly = False
+        new_object.deletable = True
+        return new_object
+
+
+def get_file_naming_script_presets():
+    """Generator of preset example file naming script objects.
+
+    Yields:
+        FileNamingScript: the next example FileNamingScript object
+    """
+    AUTHOR = N_("Picard Development Team")
```

Comment:
Note sure authors should be marked for translation.
Also I'd use `MusicBrainz Picard Development Team`.